src/marching_cubes.jl

```diff
@@ -20,18 +20,20 @@ function marching_cubes(sdf::SignedDistanceField{3,ST,FT},
     w = widths(h)
     orig = origin(HyperRectangle(sdf))
 
+    VertType, FaceType = _determine_types(MT,FT)
```

Comment:
It might be worth putting a function barrier after this line. If `_determine_types` can't be inferred for some custom mesh type, then the rest of this function will be pretty slow. But if we have a function barrier, where the inner function takes VertType and FaceType as arguments, then the inner function will be just as fast regardless of whether `_determine_types` is inferrable.

Reply:
Reading: https://docs.julialang.org/en/v1/manual/performance-tips/index.html#kernel-functions-1

If I understand correctly, it should also be valid to just put this in these calls for example, and pass the `VertType` and `FaceType` to `marching_cubes`:

```

function (::Type{MT})(df::SignedDistanceField{N,VT,FT}, method::MarchingCubes)::MT where {MT <: AbstractMesh}
    VertType, FaceType = _determine_types(MT,FT)
    marching_cubes(df, method.iso, MT, method.eps, Val(method.reduceverts), VertType, FaceType)
end
````

Reply:
So my method breaks the `marching_cubes` API unless I set defaults (which I would rather have one place in `_determine_types`). I am going to unexport the `marching_cubes` call because there is not a similar functions for the other algorithms. 

Maybe we can replace with a `extract_isosurface` function that will not be a type pirate and maybe just return a list of vertices and indices to have a more generic interface.

Reply:
Never mind, I can preserve the API.